src/collections/domain/useCases/GetCollectionItems.ts

```diff
@@ -0,0 +1,36 @@
+import { UseCase } from '../../../core/domain/useCases/UseCase'
+import { CollectionSearchCriteria } from '../../../collections'
+import { CollectionItemSubset } from '../models/CollectionItemSubset'
+import { ICollectionsRepository } from '../repositories/ICollectionsRepository'
+
+export class GetCollectionItems implements UseCase<CollectionItemSubset> {
+  private collectionsRepository: ICollectionsRepository
+
+  constructor(collectionsRepository: ICollectionsRepository) {
+    this.collectionsRepository = collectionsRepository
+  }
+
+  /**
+   * Returns an instance of CollectionItemSubset that contains reduced information for each item that the calling user can access in the installation.
+   * If the collectionId parameter is not set, the use case will return items starting from the root collection.
+   *
+   * @param {string} [collectionId] - Collection id (optional).
+   * @param {number} [limit] - Limit for pagination (optional).
+   * @param {number} [offset] - Offset for pagination (optional).
+   * @param {CollectionSearchCriteria} [collectionSearchCriteria] - Supports filtering the collection items by different properties (optional).
+   * @returns {Promise<CollectionItemSubset>}
+   */
+  async execute(
+    collectionId?: string,
```

Comment:
@GPortas could this argument be also id or alias (:number | string)? like some other use cases?

Reply:
Since the use case uses the Search API and it expects a string (for the collection alias) and not a number, the use case only supports a string as identifier: https://guides.dataverse.org/en/latest/api/search.html#search-api.

Reply:
Actually, this is tied to an implementation detail, which makes it interesting to explore in the future whether the Search API could support numeric identifiers as other use cases.